File: src/main/java/com/mrpowergamerbr/loritta/commands/nashorn/NashornMessage.java
package com.mrpowergamerbr.loritta.commands.nashorn;

import net.dv8tion.jda.core.entities.Emote;
import net.dv8tion.jda.core.entities.Message;

import java.util.List;

/**
 * Contexto de um comando Nashorn executado, é simplesmente um wrapper "seguro" para comandos em JavaScript, para que
 * a Loritta possa controlar as mensagens enviadas de uma maneira segura (para não abusarem da API do Discord)
 */
public class NashornMessage {
	private final Message message;

	public NashornMessage(Message message) {
		this.message = message;
	}

	public void editarMensagem(String texto) {
		message.editMessage(texto).complete();
	}

	public void adicionarReação(String texto) {
		List<Emote> emotes = message.getGuild().getEmotesByName(texto, false);
		if (!emotes.isEmpty()) {
			message.addReaction(emotes.get(0));
		} else {
			message.addReaction(texto).complete();
		}
	}
}

File: src/main/java/com/mrpowergamerbr/loritta/commands/nashorn/wrappers/NashornMessage.kt
package com.mrpowergamerbr.loritta.commands.nashorn.wrappers

import com.mrpowergamerbr.loritta.commands.nashorn.NashornCommand
import net.dv8tion.jda.core.entities.Emote
import net.dv8tion.jda.core.entities.Message

/**
 * Wrapper de uma mensagem de um comando Nashorn executado, é simplesmente um wrapper "seguro" para comandos em JavaScript, para que
 * a Loritta possa controlar as mensagens enviadas de uma maneira segura (para não abusarem da API do Discord)
 */
class NashornMessage(private val message: Message) {
	@NashornCommand.NashornDocs(arguments = "mensagem")
	fun editMessage(texto: String) {
		message.editMessage(texto).complete()
	}

	@NashornCommand.NashornDocs(arguments = "reação")
	fun addReaction(texto: String) {
		val emotes = message.guild.getEmotesByName(texto, false)
		if (!emotes.isEmpty()) {
			message.addReaction(emotes[0]).complete()
		} else {
			message.addReaction(texto).complete()
		}
	}
}
